android/app/src/main/java/app/organicmaps/util/Config.java

```diff
@@ -350,6 +351,49 @@ public static String getDonateUrl()
     return getString(KEY_DONATE_URL);
   }
 
+  public static void updateCounters(@NonNull Context context)
+  {
+    PreferenceManager.setDefaultValues(context, R.xml.prefs_main, false);
+
+    final SharedPreferences prefs = MwmApplication.prefs(context);
+    final SharedPreferences.Editor editor = prefs.edit();
+
+    final int launchNumber = prefs.getInt(KEY_APP_LAUNCH_NUMBER, 0);
+    editor.putInt(KEY_APP_LAUNCH_NUMBER, launchNumber + 1);
+    editor.putLong(KEY_APP_LAST_LAUNCH_TIMESTAMP, System.currentTimeMillis());
+
+    if (launchNumber == 0 || prefs.getInt(KEY_APP_FIRST_INSTALL_VERSION_CODE, 0) == 0)
+    {
+      editor.putInt(KEY_APP_FIRST_INSTALL_VERSION_CODE, BuildConfig.VERSION_CODE);
+      editor.putString(KEY_APP_FIRST_INSTALL_FLAVOR, BuildConfig.FLAVOR);
```

Comment:
Can flavor be changed now somehow without uninstalling the previous one?

Reply:
In theory, it is just a string. However, in practice, signing keys are different in different flavors. F-Droid is incompatible with Google, Google is incompatible with Web, Web is incompatible with F-Droid, etc. I am removing this option because it doesn't make sense anymore.